File: backend/src/models/Result.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/database.js';
import AudioJob from './AudioJob.js';

const Result = sequelize.define('Result', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  jobId: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: AudioJob,
      key: 'id',
    },
  },
  resultJson: {
    type: DataTypes.TEXT('long'),
    allowNull: true,
    get() {
      const rawValue = this.getDataValue('resultJson');
      return rawValue ? JSON.parse(rawValue) : null;
    },
    set(value) {
      this.setDataValue('resultJson', value ? JSON.stringify(value) : null);
    }
  },
  fileUrls: {
    type: DataTypes.TEXT('long'),
    allowNull: true,
    get() {
      const rawValue = this.getDataValue('fileUrls');
      return rawValue ? JSON.parse(rawValue) : null;
    },
    set(value) {
      this.setDataValue('fileUrls', value ? JSON.stringify(value) : null);
    }
  },
}, {
  timestamps: true,
  tableName: 'results',
});

AudioJob.hasOne(Result, { foreignKey: 'jobId' });
Result.belongsTo(AudioJob, { foreignKey: 'jobId' });

export default Result;

File: app/lib/models/result.dart
import 'audio_job.dart';

class ResultModel {
  final String id;
  final String jobId;
  final Map<String, dynamic> resultJson;
  final List<String> fileUrls;

  ResultModel({
    required this.id,
    required this.jobId,
    required this.resultJson,
    required this.fileUrls,
  });

  factory ResultModel.fromJson(Map<String, dynamic> json) {
    return ResultModel(
      id: json['id'],
      jobId: json['jobId'],
      resultJson: json['resultJson'] ?? {},
      fileUrls: List<String>.from(json['fileUrls'] ?? []),
    );
  }
}

class FullResultModel {
  final AudioJobModel job;
  final ResultModel result;

  FullResultModel({required this.job, required this.result});

  factory FullResultModel.fromJson(Map<String, dynamic> json) {
    return FullResultModel(
      job: AudioJobModel.fromJson(json['job']),
      result: ResultModel.fromJson(json['result']),
    );
  }
}
